sklearn/tree/tests/test_tree.py

```diff
@@ -2645,6 +2646,14 @@ def test_regression_tree_missing_values_toy():
     )
     assert_allclose(tree.tree_.impurity[leaves_idx], 0.0)
 
+    # With this dataset, the missing values will still be sent to the left child
+    # but the leaf will not be pure.
+    X = np.array([np.nan, 2, np.nan, 4, 5, 6]).reshape(-1, 1)
+    tree = DecisionTreeRegressor(criterion=criterion, random_state=0, max_depth=1).fit(
+        X, y
+    )
+    assert all(tree.tree_.impurity >= 0)  # MSE should always be positive
```

Comment:
Can we add a similar follow-up test for "to the right" missing values.
I think we could just parametrize the test by creating several `X` since we only check for negative MSE.

Reply:
Good point. Done!